### PR TITLE
HW-66357-V3PHP-VenmoAccount

### PR DESCRIPTION
**Pull request by jkurra-hw**

### CODE REVIEW

**Review by Rrathinasabapath — APPROVED** [review]
Verified the changes. Ready for the next review.

File: src/Hyperwallet/Hyperwallet.php [hunk 1]
```diff
@@ -299,13 +301,14 @@ public function listPaperChecks($userToken, $options = array()) {
      * @throws HyperwalletArgumentException
      * @throws HyperwalletApiException
      */
-    public function deactivatePaperCheck($userToken, $paperCheckToken) {
+    public function deactivatePaperCheck($userToken, $paperCheckToken)
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:305]: please fix code format `{`

File: src/Hyperwallet/Hyperwallet.php [hunk 2]
```diff
@@ -1803,4 +1806,202 @@ private function updateTransferMethod($userToken, $transferMethod, $transferMeth
...
+     *
+     * @param string $userToken user token
+     * @param VenmoAccount $venmoAccount Venmo account data
+     * @return VenmoAccount
+     *
+     * @throws HyperwalletArgumentException
+     * @throws HyperwalletApiException
+     */
+    public function createVenmoAccount($userToken, VenmoAccount $venmoAccount)
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:1824]: Please fix code format `{`

File: src/Hyperwallet/Hyperwallet.php [hunk 3]
```diff
@@ -1803,4 +1806,202 @@ private function updateTransferMethod($userToken, $transferMethod, $transferMeth
...
+     *
+     * @param string $userToken user token
+     * @param string $venmoAccountToken Venmo account token
+     * @return VenmoAccount
+     *
+     * @throws HyperwalletArgumentException
+     * @throws HyperwalletApiException
+     */
+    public function getVenmoAccount($userToken, $venmoAccountToken)
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:1852]: Please fix code format `{`

File: src/Hyperwallet/Hyperwallet.php [hunk 4]
```diff
@@ -1803,4 +1806,202 @@ private function updateTransferMethod($userToken, $transferMethod, $transferMeth
...
+     *
+     * @param string $userToken user token
+     * @param VenmoAccount $venmoAccount Venmo account data
+     * @return VenmoAccount
+     *
+     * @throws HyperwalletArgumentException
+     * @throws HyperwalletApiException
+     */
+    public function updateVenmoAccount($userToken, VenmoAccount $venmoAccount)
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:1877]: Please fix code format `{`

File: src/Hyperwallet/Hyperwallet.php [hunk 5]
```diff
@@ -1803,4 +1806,202 @@ private function updateTransferMethod($userToken, $transferMethod, $transferMeth
...
+     * List all Venmo accounts
+     *
+     * @param string $userToken user token
+     * @param array $options The query parameters to send
+     * @return ListResponse
+     *
+     * @throws HyperwalletApiException
+     */
+    public function listVenmoAccounts($userToken, $options = array())
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:1892]: Please fix code format `{`

File: src/Hyperwallet/Hyperwallet.php [hunk 6]
```diff
@@ -1803,4 +1806,202 @@ private function updateTransferMethod($userToken, $transferMethod, $transferMeth
...
+     *
+     * @param string $userToken user token
+     * @param string $venmoAccountToken Venmo account token
+     * @return VenmoAccountStatusTransition
+     *
+     * @throws HyperwalletArgumentException
+     * @throws HyperwalletApiException
+     */
+    public function deactivateVenmoAccount($userToken, $venmoAccountToken)
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:1913]: Please fix code format `{`

File: src/Hyperwallet/Hyperwallet.php [hunk 7]
```diff
@@ -1803,4 +1806,202 @@ private function updateTransferMethod($userToken, $transferMethod, $transferMeth
...
+     * @param string $userToken user token
+     * @param string $venmoAccountToken Venmo account token
+     * @param VenmoAccountStatusTransition $transition status transition
+     * @return VenmoAccountStatusTransition
+     *
+     * @throws HyperwalletArgumentException
+     * @throws HyperwalletApiException
+     */
+    public function createVenmoAccountStatusTransition($userToken, $venmoAccountToken, VenmoAccountStatusTransition $transition)
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:1932]: Please fix code format `{`

File: src/Hyperwallet/Hyperwallet.php [hunk 8]
```diff
@@ -1803,4 +1806,202 @@ private function updateTransferMethod($userToken, $transferMethod, $transferMeth
...
+     * @param string $userToken user token
+     * @param string $venmoAccountToken Venmo account token
+     * @param string $statusTransitionToken status transition token
+     * @return VenmoAccountStatusTransition
+     *
+     * @throws HyperwalletArgumentException
+     * @throws HyperwalletApiException
+     */
+    public function getVenmoAccountStatusTransition($userToken, $venmoAccountToken, $statusTransitionToken)
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:1959]: Please fix code format `{`

File: src/Hyperwallet/Hyperwallet.php [hunk 9]
```diff
@@ -1803,4 +1806,202 @@ private function updateTransferMethod($userToken, $transferMethod, $transferMeth
...
+     * @param string $userToken user token
+     * @param string $venmoAccountToken Venmo account token
+     * @param array $options query parameters
+     * @return ListResponse
+     *
+     * @throws HyperwalletArgumentException
+     * @throws HyperwalletApiException
+     */
+    public function listVenmoAccountStatusTransitions($userToken, $venmoAccountToken, array $options = array())
+    {
```
Comment (peter-joseph) [inline src/Hyperwallet/Hyperwallet.php:1990]: Please fix code format `{`

**Review by peter-joseph — COMMENTED** [review]
looks good to me, but fix code formatting please

File: src/Hyperwallet/Hyperwallet.php (hunk 9 above)
Comment (jkurra-hw) [reply src/Hyperwallet/Hyperwallet.php]: fixed review comment. Thanks @peter-joseph

File: src/Hyperwallet/Hyperwallet.php (hunk 4 above)
Comment (jkurra-hw) [reply src/Hyperwallet/Hyperwallet.php]: Done

File: src/Hyperwallet/Hyperwallet.php (hunk 5 above)
Comment (jkurra-hw) [reply src/Hyperwallet/Hyperwallet.php]: Done.

File: src/Hyperwallet/Hyperwallet.php (hunk 6 above)
Comment (jkurra-hw) [reply src/Hyperwallet/Hyperwallet.php]: Done

File: src/Hyperwallet/Hyperwallet.php (hunk 7 above)
Comment (jkurra-hw) [reply src/Hyperwallet/Hyperwallet.php]: Done

File: src/Hyperwallet/Hyperwallet.php (hunk 8 above)
Comment (jkurra-hw) [reply src/Hyperwallet/Hyperwallet.php]: Done

File: src/Hyperwallet/Model/VenmoAccount.php [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+ * @property \DateTime $createdOn The Venmo account creation date
+ * @property string $type The transfer method type
+ * @property string $transferMethodCountry The transfer method country
+ * @property string $transferMethodCurrency The transfer method currency
+ * @property string $isDefaultTransferMethod The is default transfer method
+ * @property string $accountId The Venmo account
+ *
+ * @package Hyperwallet\Model
+ */
+class VenmoAccount extends BaseModel
```
Comment (peter-joseph) [inline src/Hyperwallet/Model/VenmoAccount.php:19]: Please fix code format in this class

File: src/Hyperwallet/Model/VenmoAccountStatusTransition.php [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+
+namespace Hyperwallet\Model;
+
+/**
+ * Represents a V4 Venmo Account Status Transition
+ *
+ * @package Hyperwallet\Model
+ */
+class VenmoAccountStatusTransition extends StatusTransition
```
Comment (peter-joseph) [inline src/Hyperwallet/Model/VenmoAccountStatusTransition.php:11]: Please fix code format in this class

File: tests/Hyperwallet/Tests/Model/VenmoAccountTest.php [hunk 12]
```diff
@@ -0,0 +1,93 @@
+<?php
+
+
+namespace Hyperwallet\Tests\Model;
+
+
+class VenmoAccountTest extends ModelTestCase
```
Comment (peter-joseph) [inline tests/Hyperwallet/Tests/Model/VenmoAccountTest.php:7]: Please fix code format in this class

**Review by peter-joseph — COMMENTED** [review]
Please check code format please

File: src/Hyperwallet/Model/VenmoAccount.php (hunk 10 above)
Comment (jkurra-hw) [reply src/Hyperwallet/Model/VenmoAccount.php]: fixed thanks @peter-joseph

File: src/Hyperwallet/Model/VenmoAccountStatusTransition.php (hunk 11 above)
Comment (jkurra-hw) [reply src/Hyperwallet/Model/VenmoAccountStatusTransition.php]: Done.

File: tests/Hyperwallet/Tests/Model/VenmoAccountTest.php (hunk 12 above)
Comment (jkurra-hw) [reply tests/Hyperwallet/Tests/Model/VenmoAccountTest.php]: Done.

**Review by peter-joseph — APPROVED** [review]
Looks good now 👍

**Pull request merged**